orchestration/dagster_orchestration/hca_orchestration/pipelines/load_hca.py

```diff
@@ -109,9 +109,20 @@ def import_start_notification(context: HookContext) -> None:
     context.resources.slack.send_message(blocks=key_value_slack_blocks("HCA Starting Import", key_values=kvs))
 
 
-@solid
-def terminal_solid(results1: list[Optional[JobId]], results2: list[Optional[JobId]]) -> None:
-    pass
+@solid(
+    required_resource_keys={'slack', 'target_hca_dataset', 'dagit_config'}
+)
+def terminal_solid(context: HookContext, results1: list[Optional[JobId]], results2: list[Optional[JobId]]) -> None:
+    # todo: do I need to log the solid output here?
+    # context.log.info(f"Solid output = {context.solid_output_values}")
+    kvs = {
+        # todo: how do I pull staging area out here (that is from the preprocess_metadata solid config)
+        # "Staging area": context.solid_config["input_prefix"],
```

Comment:
I couldn't figure out how to obtain the staging_area needed for the load_hca completion notification. Maybe I could just store the staging_area name earlier in the script and reference that in the separate slack message blocks?

Reply:
🤔 The issue is that it's a part of a solid config rather than a globally available resource. I think the best approach would be to factor the input_prefix out of the related solid config and convert to a resource.

Reply:
Actually, we can pull it from the context after all: `context.run_config["solids"]["pre_process_metadata"]["config"]["input_prefix"],`